tests/testthat/test-merge_sce_list.R

```diff
@@ -747,9 +752,16 @@ test_that("merging SCEs with different altExps works as expected; each SCE has 1
     colnames(adt_merged),
     colnames(merged_sce)
   )
-  expect_equal(
-    sum(abs(counts(adt_merged)[, merged_sce[[batch_column]] == "sce2"])), 0
+  # check that everything is either all NA or all 0
+  all_na <- expect_true(
+    all(is.na(counts(adt_merged)[, merged_sce[[batch_column]] == "sce2"]))
   )
+  if(!all_na){
+    # will error if not all NA and not all 0
+    expect_equal(
+      sum(abs(counts(adt_merged)[, merged_sce[[batch_column]] == "sce2"])), 0
+    )
```

Comment:
I think you probably want to do this more directly for what you are trying to test?
```suggestion
  adt_counts <- counts(adt_merged)[, merged_sce[[batch_column]] == "sce2"]
  expect_true(
    all(is.na(adt_counts) | adt_counts == 0)
  )
```

But if you leave this as a dense matrix, as I think you should, then these would be properly left as `NA` to reflect that the ADTs were not measured:

```suggestion
  adt_counts <- counts(adt_merged)[, merged_sce[[batch_column]] == "sce2"]
  expect_true(
    all(is.na(adt_counts))
  )
```